neqo-transport/src/connection/tests/idle.rs

```diff
@@ -709,12 +706,6 @@ fn keep_alive_with_ack_eliciting_packet_lost() {
     let retransmit = client.process_output(now).dgram();
     assert!(retransmit.is_some());
 
-    // The next callback should be for an idle PING.
-    assert_eq!(
-        client.process_output(now).callback(),
-        IDLE_TIMEOUT / 2 - pto
```

Comment:
With this change, this timeout is actually `pto * 2`, because the idle timeout is longer.
I've removed this check since we already did the same check below.

Reply:
Maybe we need to make these checks more robust, rather than removing them.  Would this work?

```rust
    // The timeout is the smaller of:
    // * the idle timeout, less the time we've already waited (one PTO)
    // * twice the PTO, because we've already sent one probe
    assert_eq!(
        client.process_output(now).callback(),
        min(IDLE_TIMEOUT / 2 - pto, pto * 2),
    );
```

That would mean updating the code below as well.

The alternative is to sort out the relationship between the PTO timer we're using and the idle timeout, so that we can know for sure which timer is active.

Reply:
I think this should be:
```
assert_eq!(
    client.process_output(now).callback(),
    min(IDLE_TIMEOUT - pto, pto * 2),
);
```
Because we only waited one pto here.

Reply:
IDLE_TIMEOUT or IDLE_TIMEOUT/2 ?

Reply:
Should be `IDLE_TIMEOUT`.

As the test code below shows, the `start` of `IdleTimeout` is set to `now` when `_lost_packet` is sent.
We then wait for one `pto` duration to retransmit the data.
After the retransmission, `idle_timeout.expiry` returns `now + IDLE_TIMEOUT`.
Since we have only waited for one `pto` in the test, the remaining time before the idle timer expires 
is `IDLE_TIMEOUT - pto`. Therefore, in the assertion, we should use `min(IDLE_TIMEOUT - pto, pto * 2)`.


```
    // Send data on the stream that will be lost.
    _ = client.stream_send(stream, DEFAULT_STREAM_DATA).unwrap();
    let _lost_packet = client.process_output(now).dgram();

    let pto = client.process_output(now).callback();
    // Wait for packet to be marked lost.
    assert!(pto < IDLE_TIMEOUT / 2);
    now += pto;
    let retransmit = client.process_output(now).dgram();
    assert!(retransmit.is_some());
    let retransmit = client.process_output(now).dgram();
    assert!(retransmit.is_some());

    // The timeout is the smaller of:
    // * the idle timeout, less the time we've already waited (one PTO)
    // * twice the PTO, because we've already sent one probe
    assert_eq!(
        client.process_output(now).callback(),
        min(IDLE_TIMEOUT - pto, pto * 2),
    );
```

Reply:
@martinthomson , please let me know if my understanding is correct.
If you are fine with this change, I think we can land this.

Reply:
I think my suggested comment text was incorrect.

We have three operative timers:
* PTO (which is `pto * 2` at this stage)
* The idle timeout
* The keep alive timer, which is half of the idle timeout.

We can disregard the idle timeout, leaving the other two.

Reply:
Thanks, I'll update the patch and only compare the PTO.